c7n/resources/ecs.py

```diff
@@ -235,6 +237,24 @@ class ServiceTaskDefinitionFilter(RelatedTaskDefinitionFilter):
     """
 
 
+@Service.filter_registry.register('subnet')
+class SubnetFilter(net_filters.SubnetFilter):
+
+    RelatedIdsExpression = ""
+    expressions = ('taskSets[].networkConfiguration.awsvpcConfiguration.subnets[]',
+                'deployments.networkConfiguration.awsvpcConfiguration.subnets[]',
+                'networkConfiguration.awsvpcConfiguration.subnets[]')
+
+    def get_related_ids(self, resources):
+        subnet_ids = set()
+        for r in resources:
+            for exp in self.expressions:
```

Comment:
we should compile these expressions, ideally one time lazy on the instance, vs class attributes (ie avoid import time cost), else the profile on this function will be predominantly jmespath expression parsing. when using them in a tight loop.

Reply:
Makes sense! I'll make the changes shortly. Thanks!

Reply:
one other option would be just to invert the loops, iter expressions and then resources, and do a local variable compilation of the expression which should yield most of the benefit without a separate lazy attribute mechanism

Reply:
Addressed, thanks!